app/presenters/oauth_metadata_presenter.rb

```diff
@@ -61,7 +61,7 @@ def grant_types_supported
   end
 
   def token_endpoint_auth_methods_supported
-    %w(client_secret_basic client_secret_post)
+    %w(none client_secret_basic client_secret_post)
```

Comment:
We previously didn't include `none` because whilst it was _technically_ always supported, it wasn't actually supported when creating an OAuth Application, as we only supported confidential clients. The introduction of `none` here signifies that we now always support public clients.

There is a PR to doorkeeper to try to lift calculating which client authentication methods are supported up into Doorkeeper: https://github.com/doorkeeper-gem/doorkeeper/pull/1770 (though I'm not sure it'll land)

This array also exactly matches that in `TOKEN_ENDPOINT_AUTH_METHODS` in the apps controller.